scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextVariableExtractor.java

```diff
@@ -97,11 +99,9 @@
 
             // task variables from workflow definition
             if (useTaskVariables && taskContext.getInitializer().getTaskVariables() != null) {
-                for (TaskVariable taskVariable : taskContext.getInitializer().getTaskVariables().values()) {
-                    if (!taskVariable.isJobInherited()) {
-                        variables.put(taskVariable.getName(), taskVariable.getValue());
-                    }
-                }
+                variables.putAll(addTaskVariablesAndBackupInheritedVariables(variables,
+                                                                             taskContext.getInitializer()
+                                                                                        .getTaskVariables()));
```

Comment:
Instead of `variables.putAll(...)`, the method `addTaskVariablesAndBackupInheritedVariables` could directly put the variables in the given `variables` map.